mne/channels/montage.py

```diff
@@ -746,13 +764,34 @@ def _backcompat_value(pos, ref_pos):
         for name, use in zip(info_names, info_names_use):
             _loc_view = info['chs'][info['ch_names'].index(name)]['loc']
             _loc_view[:6] = _backcompat_value(ch_pos_use[use], eeg_ref_pos)
+
         del ch_pos_use
 
         # XXX this is probably wrong as it uses the order from the montage
         # rather than the order of our info['ch_names'] ...
-        info['dig'] = _format_dig_points([
+        digpoints = [
             mnt_head.dig[ii] for ii, name in enumerate(dig_names_use)
-            if name in (info_names_use + ref_names)])
+            if name in (info_names_use + ref_names)]
+
+        # get a copy of the old dig
+        if info['dig'] is not None:
+            old_dig = info['dig'].copy()
+        else:
+            old_dig = []
+
+        # determine if needed to add an extra EEG REF DigPoint
+        if custom_eeg_ref_dig:
+            # ref_name = 'EEG000' if match_case else 'eeg000'
+            ref_dig_dict = {'kind': FIFF.FIFFV_POINT_EEG,
+                            'r': eeg_ref_pos,
+                            'ident': 0,
+                            'coord_frame': info['dig'].pop()['coord_frame']}
+            ref_dig_point = _format_dig_points([ref_dig_dict])[0]
+            # only append the reference dig point if it was already
+            # in the old dig
+            if ref_dig_point in old_dig:
+                digpoints.append(ref_dig_point)
```

Comment:
It would be better not to mess with the order -- it's pretty standard to have the ref be the first of all EEG electrodes. I would use `.insert` it to the `digpoeints` list or add it to the list *first* rather than *last*. Then even the round-trip order should match, which is nice.